### PR TITLE
Parse youtube

### PR DESCRIPTION
**Pull request by nhunzaker**
Addresses #104 

You can now simply pass a youtube URL in and it should figure it out

youtubz

### CODE REVIEW

File: addons/youtube/__tests__/youtube.test.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      video_id.should.equal('')
+      done()
+    }
+
+    const component = render(<YouTube onChange={ test } />)
+
+    component._onChange({ video_id: undefined })
+  })
+
+})
```
Comment (nhunzaker) [inline addons/youtube/__tests__/youtube.test.jsx]: Any edge cases I might be missing?

File: addons/youtube/index.jsx [hunk 2]
```diff
@@ -7,6 +7,11 @@
 const Embedded = require('../common/embedded')
 const React = require('react')
 
+function parseYouTube(value='') {
+  let matches = value.match(/v=(.+)(&|$)/)
```
Comment (ltk) [inline addons/youtube/index.jsx]: YouTube share URLs can often look like `https://youtu.be/EULc7RgnM4c` as well

Comment (nhunzaker) [inline addons/youtube/index.jsx]: Boom. This is exactly what I was looking for.

Comment (nhunzaker) [inline addons/youtube/index.jsx]: Hmm. I wonder if there's a general YouTube URL parser

Comment (nhunzaker) [inline addons/youtube/index.jsx]: Check out b71783d

**Comment by mzlock** [comment]
Your label still says "YouTube Video ID" instead of URL

**Comment by mzlock** [comment]
Javascript magicks. :sparkles:  :+1:

File: addons/youtube/__tests__/youtube.test.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          video_id.should.equal('dQw4w9WgXcQ')
+          done()
+        }
+
+        const component = render(<YouTube onChange={ test } />)
+
+        component._onChange({ video_id: url })
+      })
+    }
+  })
```
Comment (nhunzaker) [inline addons/youtube/__tests__/youtube.test.jsx]: Feels on a test like this?

Comment (efatsi) [inline addons/youtube/__tests__/youtube.test.jsx]: good stuff, can't quite grasp this `done` syntax business, but that's unrelated

**Comment by nhunzaker** [comment]
@mzlock I'm not sure. I doubt that I can anticipate future YouTube URL structures, but I want to account for user error. If they save a CK block and return to editing it, it will only display the video id.

File: addons/youtube/__tests__/youtube.test.jsx (hunk 3 above)
Comment (nhunzaker) [inline addons/youtube/__tests__/youtube.test.jsx]: Ah it just means that it's an async test. The test framework, Mocha, throws an error if the test times out after a specified time period (defaulting to 2 seconds) if done isn't called.

This works because `Function.prototype.length` returns the number of parameters a function expects. So like:

```javascript
function two_params (one, two) {}

console.log(two_params.length) // => 2
```
In Mocha's case, it sees that the function length is 1, so it sets up the test as asynchronous and passes in a callback as the first argument.

**Comment by nhunzaker** [comment]
Wait... I have the power of hint text now. @mzlock @hellokv what do you think about:

screen shot 2015-08-28 at 1 57 20 pm

**Comment by nhunzaker** [comment]
You know, sans the typos

**Comment by yaychris** [comment]
:100: :gem: for this

**Comment by hellokv** [comment]
Nice. This makes the youtube embed more consistent with the other ones
we're doing.

**Comment by mzlock** [comment]
@nhunzaker - looks good with the hint text

**Comment by nhunzaker** [comment]
@mzlock cool. Thanks for the prod. This is so much better!
